docs/src/domains/troubleshoot.md

```diff
@@ -25,35 +25,164 @@ main-abcd123.abcdefgh1234567.eu.platformsh.site has address 192.0.2.1
   you might need to remove your test settings.
 
 ## Verify SSL
 
-On the command line using macOS, Linux, or the Windows Subsystem for Linux, run the following command:
+To find out where your domain is pointing to,
+you can use [the certificate checker tool](https://certcheck.pltfrm.sh/).
+This tool provides guidance on certificates,
+including when you use a [CDN](./cdn/_index.md).
+Check both the apex and the `www` domains to ensure they both point to your project.
+
+For further investigations, run the following command in a shell:
 
 ```bash
 curl -I -v  https://www.{{< variable "YOUR_DOMAIN" >}}
 ```
 
-The response should be long. Look for error messages.
-They're usually explicit enough.
-Often the problem is with a mismatch between the certificate and the domain name.
+Look for error messages.
+Often the problem is caused by a mismatch between the certificate and the domain name or an expired [custom certificate](steps/tls.md).
+
+### Error provisioning certificates
+
+When a Let's Encrypt certificate fails to provision after the build hook has completed,
+you get output similar to the following:
+
+```bash
+Provisioning certificates
+  Validating 1 new domain
+  E: Error provisioning the new certificate, will retry in the background.
+  (Next refresh will be at 2022-02-13 14:29:22.860563+00:00.)
+  Environment certificates
+W: Missing certificate for domain a-new-and-really-awesome-feature-abc1234-defghijk56789.eu3.platformsh.site
+```
+
+The renewal may fail because of the 64-character limit Let's Encrypt places on URLs.
+If you have a branch with a long name, the environment URL is over this limit and the certificate is rejected.
+To solve this, shorten your branch name so it doesn't exceed 20 characters.
+
+Generated URLs for environments have the following pattern:
+
+```bash
+{{<variable "ENVIRONMENT" >}}-{{<variable "PROJECT_ID" >}}.{{<variable "REGION" >}}.platformsh.site
+```
+
+If you have a [default domain](../define-routes/_index.md#default) set up, the generated URL has the following pattern:
+
+```bash
+{{<variable "YOUR_DOMAIN" >}}.{{<variable "ENVIRONMENT" >}}-{{<variable "PROJECT_ID" >}}.{{<variable "REGION" >}}.platformsh.site
+```
+
+The generated URLs consist of:
+
+- `{{<variable "YOUR_DOMAIN" >}}` = the amount of characters your domain has
+- `{{<variable "ENVIRONMENT" >}}` = `{{<variable "BRANCH_NAME" >}}` + 7 character hash
+- `{{<variable "PROJECT_ID" >}}` = 13 characters
+- `{{<variable "REGION" >}}` = 2 to 4 characters, depending on the region
+- `platformsh.site` = 15 characters
+- extra characters like `.` and `-` = 4 to 5 characters, depending on if you have a default domain
+
+This leaves you with 21 to 23 characters for your branch name (`{{<variable "BRANCH_NAME" >}}`) without exceeding the 64-character limit,
+depending on the region.
+To ensure your renewals succeed, 
+keep your branch names under 20 characters.
+
+### DNS Challenge
```

Comment:
❓ I think Tim had a point when he said the phrase DNS challenge could be confused with the [DNS-01 challenge](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge), which seems to have been replaced at PSH by the [HTTP-01 challenge](https://letsencrypt.org/docs/challenge-types/#http-01-challenge) => to be confirmed cf. our previous conversation. In any event, I think his suggestion to use "Let's Encrypt challenge" to avoid confusion is a good one, don't you?

Reply:
I think it's maybe not a good idea to make it too Let's Encrypt centric, to make it fit better with the CDN (and fastly) altough it's a very good point so I tried to adapt accordingly.
Please let me know if that'd work for you.